components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java

```diff
@@ -200,6 +211,35 @@ public void invoke(Request request, Response response) throws IOException, Servl
 
     }
 
+    /**
+     * This method is used to override the secured resource based on tenant-wise DCR api security configuration.
+     * @param securedResource securedResource object
+     * @param normalizedRequestURI request URL path
+     * @param httpMethod http method
+     * @param tenantDomain tenant domain of the request
+     * @throws DCRMException DCRMException
+     */
+    private void overrideSecuredResource(ResourceConfig securedResource, String normalizedRequestURI, String httpMethod,
+                                        String tenantDomain) throws DCRMException {
+
+        if (normalizedRequestURI.contains(DCR_REGISTER_ENDPOINT_PATH) && HttpMethod.POST.equals(httpMethod)) {
+
+            if (DCRMgtOGSiServiceFactory.getInstance() != null) {
+
+                DCRConfiguration dcrConfiguration = DCRMgtOGSiServiceFactory.getInstance().getDCRConfiguration();
+                Boolean isClientAuthenticationRequired = dcrConfiguration.isAuthenticationRequired();
+                if ((Boolean.TRUE).equals(isClientAuthenticationRequired)) {
```

Comment:
can simplify the if , else
securedResource.setIsSecured(Boolean.TRUE.equals(isClientAuthenticationRequired));

Reply:
This was not done because then if `isClientAuthenticationRequired` is `null` (`isClientAuthenticationRequired` is not set), `Boolean.TRUE.equals(isClientAuthenticationRequired)` value will be False, which is incorrect. 

If `isClientAuthenticationRequired` is not set, it should be the current default behaviour, where securedResource.isSecured will be True.

Reply:
In that case, we can specially handle  using the clause if(isClientAuthenticationRequired ==null)

Reply:
I think we should not handle the scenario where `isClientAuthenticationRequired` is not set, because it can change the current flow.

This is a tenant-level config and this overrides the server level configs. If tenant-level config is not set, the server level config should be there.
IS has a config in deployment.toml to configure api security for any endpoint using the endpoint url pattern. For DCR if `isClientAuthenticationRequired` is not set, that config value (if configured) will change the `securedResource.isSecured` value.